OsuPlayer.Tests/OsuPlayer.Network/ApiUserEndpointTests.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using OsuPlayer.Network.API.ApiEndpoints;
+using OsuPlayer.Network.Online;
+using Constants = OsuPlayer.Network.Constants;
+
+namespace OsuPlayer.Tests.OsuPlayer.Network;
+
+public class ApiUserEndpointTests
+{
+    [Test]
+    public async Task OfflineModeTests()
+    {
+        Constants.OfflineMode = true;
+
+        var offlineResult = await ApiAsync.GetRequestAsync<User>("nonExistingController", "nonExistingAction");
+        var offlineResult2 = await ApiAsync.ApiRequestAsync<User>("nonExistingController", "nonExistingAction", "");
+        var offlineResult3 = await ApiAsync.GetRequestWithParameterAsync<User>("nonExistingController", "nonExistingAction", "");
+        
+        Assert.IsNull(offlineResult);
+        Assert.IsNull(offlineResult2);
+        Assert.IsNull(offlineResult3);
+        
+        Constants.OfflineMode = false;
+
+        var onlineResult = await ApiAsync.GetRequestAsync<User>("nonExistingController", "nonExistingAction");
+        var onlineResult2 = await ApiAsync.ApiRequestAsync<User>("nonExistingController", "nonExistingAction", "");
+        var onlineResult3 = await ApiAsync.GetRequestWithParameterAsync<User>("nonExistingController", "nonExistingAction", "");
+        
```

Comment:
Please split into different test case. Also as mentioned before, all online tests will fail if the api is offline, but shouldn't as we have the explicit offline mode for this case, so we have to discuss practicability of these tests.

Reply:
Yeah we can discuss them I splitted them and set them on ignore